### ISSUE TITLE
/usr/bin/ld: cannot find -ltstat

### ISSUE THREAD

**Issue by 10186937** [sysml/blockmon#7]
Hi jmmlmendes,
Today is another day and another problem comes.
I download tstat-3.1.1 from http://tstat.tlc.polito.it/software.php.
After tstat is installed, I do the following steps,

```
cp [TSTATDIR]/include/libtstat.h [BLOCKMONDIR]/lib/external/tstat/
ln -s [TSTATDIR]/libtstat/.libs [BLOCKMONDIR]/lib/external/tstat/libtstatdir
```

Then do `cmake .`

```
[root@localhost blockmon]# cmake .
-- pfq (WITH_PFQ): OFF
-- packet (neither USE_SIMPLE_PACKET nor USE_SLICED_PACKET set): using old packet
-- packet tag (USE_PACKET_TAG=off): DISABLED
-- packet flow (USE_PACKET_FLOW=off): DISABLED
-- InGate queue (BLOCKING_QUEUE=off): DROPPING (i.e., non-blocking)
-- COMBO(sze2): OFF
-- TSTAT: ON
CMake Warning (dev) at CMakeLists.txt:161 (link_directories):
  This command specifies the relative path

    lib/external/tstat/libtstatdir

  as a link directory.

  Policy CMP0015 is not set: link_directories() treats paths relative to the
  source dir.  Run "cmake --help-policy CMP0015" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- daemon (WITH_DAEMON): ON
-- Configuring done
-- Generating done
-- Build files have been written to: /home/qlq/blockmon
```

and do `make`, this time the error I get is:

```
[ 99%] Building CXX object CMakeFiles/blockmoncore.dir/usr/app_twittertrending/blocks/WordCounter.cpp.o
Linking CXX shared library libblockmoncore.so
/usr/bin/ld: cannot find -ltstat
collect2: ERROR：ld return 1
make[2]: *** [libblockmoncore.so]  ERROR 1
make[1]: *** [CMakeFiles/blockmoncore.dir/all]  ERROR 2
make: *** [all]  ERROR 2

```

Anything wrong?
BTW,could you give me  a detailed operation manual or some common instances which I can run in my host directly?My email is qin.liqing@zte.com.cn.
Thanks!

**Comment by jmmlmendes** [comment]
The instructions refer to an older version of tstat.
In the latest version, .libs is located under tstat and not under libtstat (need to update the INSTALL), so you should do:
`ln -s [TSTATDIR]/tstat/.libs [BLOCKMONDIR]/lib/external/tstat/libtstatdir`

There is no manual per se but there are examples bundled with blockmon under the usr directory.
The example in the README remain the simplest one though...
After compiling, you can copy paste the example and then run:
`./blockmon XML_COMPOSITION_FILE`
to validate that it's working and start from there.

**Comment by 10186937** [comment]
Thanks jmmlmendes.
I re-installed tstat, and I got the same error.
what does `./blockmon XML_COMPOSITION_FILE` mean?
In BM shell to` start  XML_COMPOSITION_FILE `?

**Comment by jmmlmendes** [comment]
The XML composition file is what defines the interconnections between blocks and the block's configuration (i.e. number of threads, block-specific parameters, ...).

You can execute the blockmon binary directly giving this file as an argument:
`./blockmon XML_COMPOSITION_FILE`
 or you can use the blockmon CLI/daemon to start compositions.

**Comment by jmmlmendes** [comment]
Regarding the error, did you re-run `cmake .` after creating the symlink?

**Comment by 10186937** [comment]
Yes, I will re-run `cmake -DWITH_<FEATURE>=[ON/OFF] `& `cmake .` whenever I do some changes.

**Comment by 10186937** [comment]
Hi jmmlmendes, Could I say something else?
How many application scenarios does blockmon support now? 
Are you still working on new functionality and May I know your plans?
Thank you very much.

**Comment by 10186937** [comment]
Hi jmmlmendes,
In addition to the points mentioned above, does blockmon a great help in an openstack environment?
Does blockmon support multi-nodes?
Thanks!

**Comment by jmmlmendes** [comment]
Hi,
Since these are generic questions (not related to this issue) would be better if you contacted blockmon@neclab.eu
Have you tried contacting through this email address?

Regarding the current issue, did you manage to link with tstat and run blockmon?
